### PR TITLE
[ACS-4075] - Folder Rules bugfix: Execute Script uses node ref picker instead of dropdown

### PR DESCRIPTION
**Pull request by nikita-web-ua**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

 Execute Script action uses node ref picker instead of dropdown while editing the rule.
https://alfresco.atlassian.net/browse/ACS-4075

**What is the new behaviour?**

`parameterConstraints$` were loading every time the aca-edit-rule-dialog opened up, so in case of rule editing `constraintsForDropdownBox` was `undefined` that is why node selector was displayed instead of dropdown box. Now aca-edit-rule-dialog redesigned from smart to ui component so  `actionDefinitions$` and `parameterConstraints$` are loaded once on aca-manage-rules component.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.ts [hunk 1]
```diff
@@ -64,8 +65,10 @@ export class ManageRulesSmartComponent implements OnInit, OnDestroy {
 
   actionsLoading$: Observable<boolean>;
   actionDefinitions$: Observable<ActionDefinitionTransformed[]>;
+  parameterConstraints$: Observable<ActionParameterConstraint[]>;
 
   private destroyed$ = new Subject<void>();
+  private _actionDefinitionsSub;
```
Comment (MichalKinas) [inline projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.ts:71]: Can we define type of this variable upfront?

File: projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.ts [hunk 2]
```diff
@@ -103,11 +107,16 @@ export class ManageRulesSmartComponent implements OnInit, OnDestroy {
         this.folderRuleSetsService.loadRuleSets(this.nodeId);
       }
     });
+
+    this._actionDefinitionsSub = this.actionDefinitions$.subscribe((actionDefinitions) =>
```
Comment (MichalKinas) [inline projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.ts:111]: Same here can we specify type of `actionDefinitions`

File: projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.ts (hunk 1 above)
Comment (nikita-web-ua) [reply projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.ts]: Added, thanks.

File: projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.ts (hunk 2 above)
Comment (nikita-web-ua) [reply projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.ts]: Done, ty.

**Review by MichalKinas — APPROVED** [review]

**Review by AleksanderSklorz — APPROVED** [review]

**Review by DenysVuika — APPROVED** [review]

**Pull request merged**